### PR TITLE
Improved collector: uniqueness of user/connector pairs and filtering of unusable pairs

### PR DESCRIPTION
**Pull request by konstan**
Improvements/changes

- uniqueness of user/connector pairs in the local bookkeeping atom;
- filtering out of unusable user/connector pairs; added unit test;
- improved logging;
- added boot's build.boot  (Leinegen now is failing in REPL).

Connected to #795

### CODE REVIEW

File: jar-async/src/slipstream/async/collector.clj [hunk 1]
```diff
@@ -159,53 +162,77 @@
...
+        (busy! u c)
+        (>! chan [u c])
+        (inform-inserted context u c t))
+      (inform-avoiding context u c))))
+
+(defn compose-uc-pairs
+  [users connectors]
+  (remove nil?
+          (for [u users c connectors]
+            (when (.isCredentialsSet c u) [u c]))))
```
Comment (st) [inline jar-async/src/slipstream/async/collector.clj]: minor:
`remove nil?` can be avoided by using `:when` inside `for`
 (for [u users c connectors] :when (.isCredentialsSet c u) [u c]))

Comment (konstan) [inline jar-async/src/slipstream/async/collector.clj]: Thanks! Updating.

File: jar-async/src/slipstream/async/collector.clj [hunk 2]
```diff
@@ -43,21 +46,21 @@
...
 
 (defn- busy!
   [u c]
-  (swap! busy mark-busy [(get-name u) c]))
+  (swap! busy mark-busy [(get-name u) (get-conn-inst-name c)]))
 
 (defn- free!
   [u c]
-  (swap! busy mark-free [(get-name u) c]))
+  (swap! busy mark-free [(get-name u) (get-conn-inst-name c)]))
```
Comment (st) [inline jar-async/src/slipstream/async/collector.clj]: suggestion:
factor similar calls  [(get-name u) (get-conn-inst-name c)]
in
```
(defn- tuple-uc
  [u c]
  [(get-name u) (get-conn-inst-name c)])
```

File: jar-async/src/slipstream/async/collector.clj [hunk 3]
```diff
@@ -32,6 +31,10 @@
   [user]
   (.getName user))
 
+(defn get-conn-inst-name
+  [connector]
```
Comment (st) [inline jar-async/src/slipstream/async/collector.clj]: can be private?

File: jar-async/src/slipstream/async/collector.clj [hunk 4]
```diff
@@ -91,7 +94,7 @@
 
 (defn- user-connector
  [user connector]
- (apply str "[" (get-name user) "/" (.getConnectorInstanceName connector) "] " (System/currentTimeMillis) " "))
+ (apply str "[" (get-name user) "/" (get-conn-inst-name connector) "] " (System/currentTimeMillis) " "))
```
Comment (st) [inline jar-async/src/slipstream/async/collector.clj]: Could be replaced by `tuple-uc` ?

File: jar-async/src/slipstream/async/collector.clj (hunk 1 above)
Comment (konstan) [inline jar-async/src/slipstream/async/collector.clj]: Pushed update (with a minor change.)

File: jar-async/src/slipstream/async/collector.clj (hunk 2 above)
Comment (konstan) [inline jar-async/src/slipstream/async/collector.clj]: Indeed.

**Comment by konstan** [comment]
@st  Addressed your comments.  Please check.  Thanks.

**Pull request merged**